docs/source/testing.rst

```diff
@@ -0,0 +1,316 @@
+explicitly create a mechanism turning a List of ``Requirement`` objects into something
+that can be tested by pytest manually. Importantly, we want every ``Constraint`` of every
+``Requirement`` to be tested independently of each other. For instance, we would not like
+one failing test to halt all others.
+
+Many of these approaches rely on adapting pytest's ``conftest.py``. If you are not familiar
+with this concept, you might want to read up on it
+`here <https://docs.pytest.org/en/6.2.x/writing_plugins.html#conftest-py-plugins>`_.
+
+Subselection
+------------
+
+Most often one might want to run all tests implied by a specification.
+
+Yet, for example after believing to have fixed a data problem, one might simply
+want to test whether a single test, which had previously been failing, succeeds
+at last.
+
+Another example for when one would like to test a subset of tests is if the data at
+hand is not available in its entirety. Rather, it could be that one would like
+to run a subset of the test suite against a subsample of the typical dataset.
+
+In this section we present to approaches to do subselection of tests.
+
+Ex-post: subselecting generated tests
+*************************************
+
+Instead of merely running  ``$ pytest specification.py`` one may add pytests's
+``-k`` flag and specify the ``Constraint`` (s) one cares about.
+
+Importantly, every ``Constraint`` object can be identified via a name. If one wants
+to figure out how this string is built, please refer to the implementation of
+:meth:`~datajudge.constraints.base.Constraint.get_description`.
+Otherwise, one could also just run all of the tests once and investigate
+the resulting test report to find the relevant names.
+
+When only caring about the ``UniquesEquality`` constraint in our
+:doc:`twitch example <examples/example_twitch>`.
+one might for instance use the following prefix the filter for it:
+
+.. code-block:: console
+
+  $ pytest twitch_specification.py -k "UniquesEquality::public.twitch_v1"
+
+Ex-ante: Defining categories of tests
+*************************************
+
+Another option to subselect a certain set of tests is by use of
+`pytest markers <https://docs.pytest.org/en/7.1.x/example/markers.html>`_.
+The following is one way of using markers in conjunction with ``datajudge``.
+
+In this particular illustration we'll allow for two markers:
+
+* ``basic``: indicating that only truly fundamental tests should be run
+* ``all``: indicating that any available test should be run
+
+For that matter we'll add a bit of pytest magic to the respective ``conftest.py``.
+
+.. code-block:: python
+  :caption: ``conftest.py``
+
+  def pytest_generate_tests(metafunc):
+      if "basic_constraint" in metafunc.fixturenames:
+           metafunc.parametrize(
+	       "basic_constraint",
+               metafunc.module.get_basic_constraints(),
+               ids=metafunc.module.idfn,
```

Comment:
You're invoking this `idfn` callable, but in this local context, the one of this example, it's never defined.

Reply:
Nice catch :)